pytools/tag.py

```diff
@@ -257,4 +262,152 @@ def without_tags(self: T_co,
 
 # }}}
 
+
+# {{{ parse
+
+TAG_GRAMMAR = """
+    tag: tag_class "(" params ")" -> map_tag_from_python_class
+       | SHORTCUT                 -> map_tag_from_shortcut
+
+    params:                       -> map_empty_args_params
+          | args                  -> map_args_only_params
+          | kwargs                -> map_kwargs_only_params
+          | args "," kwargs       -> map_args_kwargs_params
+
+    ?kwargs: kwarg
+           | kwargs "," kwarg     -> map_kwargs
+
+    args: arg                     -> map_singleton_args
+        | args "," arg            -> map_args
+
+    kwarg: name "=" arg           -> map_kwarg
+
+    ?arg: tag
+        | INT                     -> map_int
+        | ESCAPED_STRING          -> map_string
+
+    tag_class: module "." name    -> map_tag_class
+
+    module: name                  -> map_top_level_module
+          | module "." name       -> map_nested_module
+
+    name: CNAME                   -> map_name
+    SHORTCUT: "." ("_"|LETTER) ("_"|LETTER|DIGIT|".")*
+
+    %import common.INT
+    %import common.ESCAPED_STRING
+    %import common.DIGIT
+    %import common.LETTER
+    %import common.CNAME
+    %import common.WS
+    %ignore WS
+"""
```

Comment:
This should probably be a separate `.lark` file so that it's importable elsewhere.

https://lark-parser.readthedocs.io/en/latest/grammar.html#import

Reply:
(Unless it's importable this way? The docs didn't make it sound that way.)

Reply:
If it becomes a `.lark` file, make sure to also add that to `MANIFEST.in`.

Reply:
Yup, done. See 6583c32b3c74745803243c0e4df0d2d96cf32a54.